src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,38 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            mainDiagonal[i] = matrix[i][i];
+        }
+        return mainDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] secondaryDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            secondaryDiagonal[i] = matrix[i][matrix.length - 1 - i];
+        }
+        return secondaryDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+        for (int[] row : matrix) {
+            for (int cell : row) {
+                min = (cell < min) ? min = cell : min;
```

Comment:
It doesn't make sense to reassign the same value again (`min = min`). Consider using the `if` statement.

Reply:
Thought to practice with tenary. Agree and fix is done.